### PR TITLE
OSP Compatibility Updates

### PR DESCRIPTION
**Pull request by jarretlavallee**
This PR has some basic OSP compatibility updates to the module. It includes the following. 

* Use the client certificate for all PuppetDB API requests
* ~~Update the `pe_server_version` conditionals to check if `NotUndef`~~
* Remove the file sync query from PuppetDB when `pe_server_version` is not defined
* Allow for specifying the PostgreSQL databases in the profile
* Failing on the `postgres_access` class if it is OSP since it uses PE classes and we cannot guarantee the method used in OSP.

The biggest change is the client certificates on the PuppetDB API requests over skipping validation.

This is a draft PR since there are no spec tests on these profiles and want to identify what is appropriate.

### CODE REVIEW

File: manifests/profile/compiler.pp [hunk 1]
```diff
@@ -37,16 +37,18 @@
...
-    options     => [{
-      'urls'                 => [ "https://${compiler}:${port}/status/v1/services/file-sync-client-service?level=debug" ],
-      'insecure_skip_verify' => true,
-      'data_format'          => 'json',
-      'json_string_fields'   => ['status_repos_puppet-code_latest_commit_date'],
-      'timeout'              => $timeout,
-    }],
-    notify      => Service['telegraf'],
-    require     => Package['telegraf'],
+if $facts['pe_server_version'] =~ NotUndef {
```
Comment (genebean) [inline manifests/profile/compiler.pp]: I think you should just do this in all the places you reference this fact:

```suggestion
if $facts['pe_server_version'] {
```

File: manifests/profile/master/postgres.pp [hunk 2]
```diff
@@ -18,6 +18,7 @@
   Variant[String,Tuple[String, Integer]] $postgres_host = $facts['networking']['fqdn'],
   String[2] $query_interval                             = lookup('puppet_metrics_dashboard::pg_query_interval'),
   Integer[1] $port                                      = 5432,
+  Array[String] $databases                              = ['pe-puppetdb','pe-rbac','pe-activity','pe-classifier'],
```
Comment (genebean) [inline manifests/profile/master/postgres.pp]: Please set this to be an array of non-empty strings by doing something like this:

```suggestion
  Array[String[1]] $databases                              = ['pe-puppetdb','pe-rbac','pe-activity','pe-classifier'],
```

Also, please add docs to the comments section above for this new parameter.

File: manifests/profile/puppetdb.pp [hunk 3]
```diff
@@ -28,18 +31,39 @@
   Puppet_metrics_dashboard::Puppetdb_metric $puppetdb_metrics = puppet_metrics_dashboard::puppetdb_metrics(),
   Integer[1] $port                                            = 8081,
   String[2] $interval                                         = '5s',
+  Boolean $enable_client_cert                                 = true,
   ){
 
+  if ! defined(Puppet_metrics_dashboard::Certs['telegraf']) {
```
Comment (genebean) [inline manifests/profile/puppetdb.pp]: I suggest using stdlib's ensure_resource for this

File: manifests/profile/compiler.pp (hunk 1 above)
Comment (jarretlavallee) [reply manifests/profile/compiler.pp]: Sounds good to me.

File: manifests/profile/master/postgres.pp (hunk 2 above)
Comment (jarretlavallee) [reply manifests/profile/master/postgres.pp]: Done!

File: manifests/profile/puppetdb.pp (hunk 3 above)
Comment (jarretlavallee) [reply manifests/profile/puppetdb.pp]: Great idea! I changed both locations it is used.

**Comment by suckatrash** [comment]
@jarretlavallee - could you also update changelog with this PR?

**Comment by jarretlavallee** [comment]
Thanks @suckatrash. The changelog has been updated. It put it as a Y release with the change to the client ssl certs. Let me know if you think it would qualify as a Z release instead.

**Comment by suckatrash** [comment]
Makes sense to me. @genebean anything else needed here from your perspective?

**Pull request merged**